sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.BlobAccessConditions;
+import com.azure.storage.blob.models.DeleteSnapshotsOptionType;
+import com.azure.storage.blob.models.LeaseAccessConditions;
+import com.azure.storage.common.Constants;
+import com.azure.storage.common.policy.SharedKeyCredentialPolicy;
+import reactor.core.Disposable;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.context.Context;
+
+import java.net.MalformedURLException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This class allows for batching of multiple Azure Storage operations in a single request via
+ * {@link BlobBatchClient#submitBatch(BlobBatch)} or {@link BlobBatchAsyncClient#submitBatch(BlobBatch)}.
+ *
+ * <p>Creating a batch requires either a {@link BlobServiceClient} or {@link BlobServiceAsyncClient}.</p>
+ *
+ * {@codesnippet com.azure.storage.blob.BlobBatch.createWithServiceClient}
+ * {@codesnippet com.azure.storage.blob.BlobBatch.createWithServiceAsyncClient}
+ *
+ * <p>Azure Storage Blob batches are homogeneous which means a {@link #delete(String) delete} and {@link
+ * #setTier(String, AccessTier) set tier} are not allowed to be in the same batch.</p>
+ *
+ * {@codesnippet com.azure.storage.blob.BlobBatch.illegalBatchOperation}
+ *
+ * <p>Please refer to the <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch">Azure Docs</a>
+ * for more information.</p>
+ */
+public final class BlobBatch {
+    private static final String BATCH_REQUEST_CONTENT_ID = "Batch-Request-Content-Id";
+    private static final String BATCH_REQUEST_URL_PATH = "Batch-Request-Url-Path";
+    private static final String CONTENT_ID = "Content-Id";
+    private static final String BATCH_BOUNDARY_TEMPLATE = "batch_%s";
+    private static final String CONTENT_TYPE = "Content-Type: application/http";
+    private static final String CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding: binary";
+    private static final String CONTENT_ID_TEMPLATE = "Content-ID: %d";
+    private static final String HTTP_VERSION = "HTTP/1.1";
+    private static final String OPERATION_TEMPLATE = "%s %s %s";
+    private static final String HEADER_TEMPLATE = "%s: %s";
+    private static final String NEWLINE = "\r\n";
+
+    private static final int[] EXPECTED_DELETE_STATUS_CODES = { 202 };
+    private static final int[] EXPECTED_SET_TIER_STATUS_CODES = { 200, 202 };
+
+    private final ClientLogger logger = new ClientLogger(BlobBatch.class);
+
+    private final BlobAsyncClient batchClient;
```

Comment:
>batchClient [](start = 34, length = 11)

Can you rename this to blobClient? Calling it batchClient feels misleading (and makes reading some of the code below confusing)

Reply:
Done